### PR TITLE
CI Visibility Maven instrumentation

### PR DESCRIPTION
**Pull request by nikita-tkachenko-datadog**
# What Does This Do
Adds a new instrumentation for Maven build processes.
The instrumentation creates spans for entire build and for each surefire/failsafe plugins goal within a build.
The instrumentation passes IDs of created spans to forked test processes (so that the spans created in those processes could refer to the spans from the parent daemon process).

# Motivation
This is needed in order to support Test Session and Test Module levels visibility (a CI Visibility feature) for Maven builds.
"Test session" is a CI Visibility term that is equivalent to tests run for a single project, "Test module" is equivalent to tests run for a single module within a project.

### CODE REVIEW

**Review by devinsba — APPROVED** [review]
Looks good to me

**Review by albertvaka — COMMENTED** [review]
LGTM

File: dd-java-agent/instrumentation/maven-3.2.1/src/main/java/datadog/trace/instrumentation/maven3/MavenUtils.java [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+  /**
+   * Returns command line used to start the build. Depending on Maven version the actual command
+   * line might not be available, in which case we will do our best to recreate one using the state
+   * of the request object
+   */
+  public static String getCommandLine(MavenSession session) {
+    String mavenCmdLineArgsEnvVar = System.getenv(MAVEN_CMD_LINE_ARGS_ENVIRONMENT_VAR);
+    if (mavenCmdLineArgsEnvVar != null) {
+      return "mvn" + mavenCmdLineArgsEnvVar;
```
[the remark this answers is not in the document]
Comment (nikita-tkachenko-datadog) [reply dd-java-agent/instrumentation/maven-3.2.1/src/main/java/datadog/trace/instrumentation/maven3/MavenUtils.java:38]: yep, the environment var begins with a whitespace, so there's no need to add one manually

File: dd-java-agent/instrumentation/maven-3.2.1/src/main/java/datadog/trace/instrumentation/maven3/MavenUtils.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    }
+
+    Properties sessionSystemProperties = session.getSystemProperties();
+    String mavenCmdLineArgsProp =
+        sessionSystemProperties.getProperty("env." + MAVEN_CMD_LINE_ARGS_ENVIRONMENT_VAR);
+    if (mavenCmdLineArgsProp != null) {
+      return MVN_CMD_LINE_INVOCATION + mavenCmdLineArgsProp;
```
[the remark this answers is not in the document]
Comment (nikita-tkachenko-datadog) [reply dd-java-agent/instrumentation/maven-3.2.1/src/main/java/datadog/trace/instrumentation/maven3/MavenUtils.java:45]: Good catch! I was too quick to merge this after Brian's approval, but I'll be sure to address this in an upcoming PR

**Pull request merged**